legal-api/src/legal_api/models/business.py

```diff
@@ -563,6 +564,21 @@ def get_filing_by_id(cls, business_identifier: int, filing_id: str):
             one_or_none()
         return None if not filing else filing[1]
 
+    @classmethod
+    def is_pending_amalgamating_business(cls, business_identifier: int):
+        """Check if a business has a pending amalgamation with the provided business identifier. """
+        # Construct a JSON containment check clause for the SQL query
+        where_clause = text(
```

Comment:
@argush3 if we face performance issue for this we can add index for this search. https://stackoverflow.com/questions/18404055/index-for-finding-an-element-in-a-json-array/18405706#18405706

Reply:
oh nice!